org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/MavenClasspathHelpers.java

```diff
@@ -80,4 +80,9 @@ public static boolean hasTestFlagDisabled(MavenProject mavenProject) {
     return mavenProject != null
         && Boolean.valueOf(mavenProject.getProperties().getProperty("m2e.disableTestClasspathFlag", "false"));
   }
+
+  public static boolean isTestSkipped(MavenProject mavenProject) {
+    return mavenProject != null
+        && Boolean.valueOf(mavenProject.getProperties().getProperty("maven.test.skip", "false"));
```

Comment:
This only checks if the `maven.test.skip` property is defined in the properties section of the current project.
It does not check if that property is set in the parent or is configured in the [maven-compiler-plugin's `testCompile` goal](https://maven.apache.org/plugins/maven-compiler-plugin/testCompile-mojo.html#skip).

According to https://maven.apache.org/plugins-archives/maven-surefire-plugin-2.12.4/examples/skipping-test.html:
`maven.test.skip is honored by Surefire, Failsafe and the Compiler Plugin.`

I'm not yet sure if it is useful to check [maven-surefire-plugin:skip](https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#skip) too, because I don't expect the test goal to trigger the compiler-plugin again.
But I haven't checked that.

Therefore I suggest to delete this method and instead check in the caller if that configuration property is set.
You could do that by creating a new flag `boolean isTestCompilationSkipped = false;` in `AbstractJavaProjectConfigurator.addProjectSourceFolders()` just before the loop that iterates over all `executions` and only considers `executions` for which `isTestCompileExecution` returns true.
Within that loop you could simply set the flag to the configured value with
`isTestCompilationSkipped = maven.getMojoParameterValue(mavenProject, compile, "skip", Boolean.class, monitor);`

This should also cover the cases mentioned above.

Reply:
Done